src/main/java/dev/lu15/voicechat/VoiceChatImpl.java

```diff
@@ -46,19 +43,34 @@ final class VoiceChatImpl implements VoiceChat {
     private final @NotNull VoiceServer server;
     private final int port;
     private final @NotNull String publicAddress;
+    private final int mtu;
+    private final Codec codec;
+    private final boolean groups;
+    private final boolean recording;
+    private final int distance;
+    private final int keepalive;
+
+    private final GroupManager groupManager = new GroupManager();
+
 
     @SuppressWarnings("PatternValidation")
-    private VoiceChatImpl(@NotNull InetAddress address, int port, @NotNull EventNode<Event> eventNode, @NotNull String publicAddress) {
+    private VoiceChatImpl(@NotNull InetAddress address, int port, @NotNull String publicAddress, @NotNull EventNode<Event> eventNode, int mtu, Codec codecc, boolean groups, int distance, int keepalive, boolean recording) {
```

Comment:
This constructor is probably big enough to put each parameter on a separate line. Also, `codecc` -> `codec`.